### ISSUE TITLE
Set username dynamically from previous request

### ISSUE THREAD

**Issue by apparentorder** [Orange-OpenSource/hurl#1868]
### Problem to solve

As follow-up to #1844 (thanks!) –

I'd like to perform SigV4 signed requests with tokens / temporary credentials. This usually involves two steps:

1. Request to get the credentials / token (e.g. from AWS STS or Cognito Identity Pool)
1. Actual request(s), signed with the credentials retrieved in step 1

Currently I'd have to do this as two `hurl` calls in a wrapper script.

### Proposal

This would be easy to achieve if username/password could be set from variables, for example
```
POST request#1
[Captures]
session_access_key_id: jsonpath "$.foo"
session_secret_key: jsonpath "$.bar"

POST request#2
[Options]
username: {{session_access_key_id}}
password: {{session_secret_key}}
aws-sigv4: aws:amz:foo:bar
```

I imagine this could be implemented easily (I could try to come up with a PR), but I don't fully grasp potential side effects on `[BasicAuth]` yet (see also #898).

This would also allow to set username/password from `HURL_foo` environment variables, which improves security (no credentials in the shell history and in system-wide process list).

What would be great additionally: If variables persisted across hurlfiles, so that I could use `hurl --test login.hurl request1.hurl request2.hurl`, as maintaining a large number of tests in one hurlfile feels cumbersome (might be worth a separate issue).

**Comment by fabricereix** [comment]
Supporting dynamic field name for the `[BasicAuth]` section (as described in #1877) will probably be a good solution.

**Comment by jcamiel** [comment]
Hi @apparentorder the feature has been merged in #2012 

Now this file is supported:

```
POST https://foo.com
[Options]
variable: user=bob
variable: secret=foo
[BasicAuth]
{{user}}: {{secret}}
HTTP 200
```
